### DISCUSSION TITLE
Project Map Data to Cross Section Error - ArcGIS Pro 3.3.1

### DISCUSSION THREAD

**Discussion by kxwhipple** [usgs/gems-tools-pro#116] [Q&A]
I have built a simple mapping geodatabase (Create New Database) to test the Cross Section tool in ArcGIS Pro 3.3.1. The Tool is failing immediately with an error that suggests either there is a minor glitch in the code or something simple wrong with my cross-section line feature class. The error occurs immediately and is one that was discussed and fixed in a prior version in late Spring of 2022: 
Checking section line
  Making feature data set CrossSectionC_test
  Prepping section line
    copying XSectionC to xxxXsLine
  File "<expression>", line 1
    01
    ^
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

Has anyone else seen this problem? Any suggestions? Anyone using Project Map Data to Cross Section without trouble in ArcGISPro 3.3.1?

**Comment by vlatane-va-energy** [comment]
Does your cross section line have a CartographicLines_ID field, and is it filled out with "prefix" + "integer"? I believe that's where I've seen this error crop up before. Not yet tested in 3.3.1, only 3.2 or lower.

**Reply by kxwhipple** [reply]
Thanks, but no. Section lines are very simple. I've tried ones with only Object ID, Shape, Shape_Length or Object ID, Shape, SectionID (text), Shape_Length with the same result.  My first section line feature class (PolyLine) had the SectionID field (text) with value set to 1, so I thought that might be the problem, so changed it to 10, with no affect. Then created a new section line without even this field, same result.  I should have mentioned I am using the latest version of GeMS Toolbox, downloaded 8-25-24, v. 2.13.2.

**Reply by vlatane-va-energy** [reply]
Sorry, should have clarified. I have seen best results when the CartographicLines_ID field _is_ filled out with a "prefix" + "integer", e.g. CAL01. Good luck!

**Comment by ethoms-usgs** [comment]
@kxwhipple, it looks like you are not using the latest version of the tool which is only on the [xs-update](https://github.com/DOI-USGS/gems-tools-pro/tree/xs-update) branch of the toolbox. Be sure to download from the green Code button, not the Releases section and ignore warnings about using an obsolete version.

I apologize for this not being more clear. It's probably time I merge this tool with the main branch.

**Reply by kxwhipple** [reply]
Thanks for the helpful and quick reply!  I definitely missed that, assumed that fix from Spring 22 was in the latest release.  Switching to the updated version I get past that original error about syntax and leading zeroes, but unfortunately hit a new error in several attempts (and after restarting ArcPro to make sure its using the updated xs tool):

  File "C:\GEMS_USGS_Mapping\gems-tools-pro-2.13.2\Scripts\GeMS_ProjectCrossSectionData.py", line 346, in <module>
    height = 0 if sys.argv[14] in nulls else int(sys.argv[14])
                  ~~~~~~~~^^^^
IndexError: list index out of range

Does suggest some expected setting is different in my geodatabase.

Note: I replaced the python script in my toolbox folder, rather than replacing the whole toolbox as I didnt want to lose any other updates.  The error message confirms I have the right one as it warns I am using an obsolete version, and that I am using: GeMS_ProjectCrossSectionData_Arc10.py, version of 20 April 2022

**Reply by ethoms-usgs** [reply]
Although you are using the correct version of the script, it expects more parameters, so you do need to use the new form for the tool in the xs-update toolbox. For now, use that version of the toolbox only for the cross-section tool.

**Reply by kxwhipple** [reply]
Thanks again. I really appreciate the help and quick responses. My mistake to assume it was only the *.py for this function that needed to be replaced.  Added the full xs-update toolbox and tried it for the Cross Section Tool.  It ran for longer, made progress, but exited with a new error:

 ﻿Traceback (most recent call last):
﻿  File "C:\GEMS_USGS_Mapping\gems-tools-pro-xs-update\Scripts\GeMS_ProjectCrossSectionData_AGP2.py", line 489, in <module>
﻿    desc = arcpy.da.Describe(fc)
﻿           ^^^^^^^^^^^^^^^^^^^^^
﻿ValueError: Object: Error in accessing describe

For this simple test I have only a MapBorder, ContactsandFaults, and MapUnitPolys with data. Topology is good for Contacts and Faults and the Map polygons constructed from them. As such in the Cross section tool I am projecting only ContactsandFaults and MapUnitPolys rather than checking the Project all Features in GeologicMap option, just FYI.

**Comment by ethoms-usgs** [comment]
Ok, I have an idea of what's going on. Can you try this?
Replace the script file you have with the one I have attached (remove the .txt extension)
[GeMS_ProjectCrossSectionData_AGP2.py.txt](https://github.com/user-attachments/files/16765829/GeMS_ProjectCrossSectionData_AGP2.py.txt)

**Comment by kxwhipple** [comment]
Unfortunately I get the same error:
﻿  File "C:\GEMS_USGS_Mapping\gems-tools-pro-xs-update\Scripts\GeMS_ProjectCrossSectionData_AGP2.py", line 488, in <module>
﻿    desc = arcpy.da.Describe(fc)
﻿           ^^^^^^^^^^^^^^^^^^^^^
﻿ValueError: Object: Error in accessing describe

The run does produce the CrossSection feature dataset, which holds three Polylines - a copy of the cross section line (CSname_sectionline, CSname_z, and CSname_zm), but attribute tables of all three are the same - just the line, no elevations, only Shape differs (Polyline, Polyline Z, Polyline ZM). Maybe something is off with my geodatabase?

**Reply by ethoms-usgs** [reply]
Odd. I can't generate the error unless I pass in a fake name and the update I sent you sends a catalog path instead of just the name, to make it more clear where the feature class is.

Can you send me your geodatabase?

**Comment by kxwhipple** [comment]
Evan -

I've learned more about behavior. I've been talking to Roman DiBiase and Erin DiMaggio and a sample geodatabase of theirs worked fine using the gems-tools-pro-xs-update (both with your original xs-update script and one that Roman edited to correct the apparent dip issue), but only with certain settings in the tool.

I now find that in my test geodatabase both the original xs-update script and Roman's slightly edited version both work but ONLY IF I select "Project all features in GeologicMap" (prior I was not doing this but rather pointing to only ContactsandFaults and MapUnitPolys because all other features are empty in my simple test - leaving this unchecked and specifying only these layers results in the error I sent before) and ONLY IF I select "Save intermediate data" (with this unchecked, the script proceeds pretty far but ultimately fails).

Interestingly the updated version you posted this afternoon is generating a new error for me:
  File "C:\GEMS_USGS_Mapping\gems-tools-pro-xs-update\Scripts\GeMS_ProjectCrossSectionData_AGP2.py", line 505, in <module>
﻿    arcpy.analysis.Intersect([zm_line, fc_path], intersect_pts, "ALL", None, "POINT")
﻿  File "C:\Users\kwhipple\AppData\Local\Programs\ArcGIS\Pro\Resources\ArcPy\arcpy\analysis.py", line 511, in Intersect
﻿    raise e
﻿  File "C:\Users\kwhipple\AppData\Local\Programs\ArcGIS\Pro\Resources\ArcPy\arcpy\analysis.py", line 508, in Intersect
﻿    retval = convertArcObjectToPythonObject(gp.Intersect_analysis(*gp_fixargs((in_features, out_feature_class, join_attributes, cluster_tolerance, output_type), True)))
﻿                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
﻿  File "C:\Users\kwhipple\AppData\Local\Programs\ArcGIS\Pro\Resources\ArcPy\arcpy\geoprocessing\_base.py", line 512, in <lambda>
﻿    return lambda *args: val(*gp_fixargs(args, True))
﻿                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
﻿arcgisscripting.ExecuteError:  ERROR 000210: Cannot create output C:\Users\kwhipple\ASU Dropbox\Kelin Whipple\glg362\ArcPro_Alluvial_Fan_test\Alluvial_Fan_Template\Alluvial_Fan_Mapping.gdb\CrossSectionET_all_settings\CSET_all_settingsXSectionA_intersections
Failed to execute (Intersect).

So it IS working without edits from the xs-update version, so long as those two options are selected.

If you's still like to take a look at my project (which would be great), what is the best way to share my project? - I have both a mapping geodatabase and what you might call a larger basemaps geodatabase with the dem and other rasters for the project, but I could send just the DEM for this.

Kelin

________________________________

**Reply by ethoms-usgs** [reply]
Yes, I would like to take a look. Send me the mapping geodatabase. It should be fine to remove all feature classes aside from the lines and polygons. I will generate a rough dem here. Email is probably the best way but do whatever you think works best.

**Comment by kxwhipple** [comment]
Thanks for taking a look.
First attempt to attach as a zip archive was rejected by the system - below I provide a dropbox link to the zipped geodatabase.

You'll see its full of CrossSection feature datasets from all my testing with different versions of the py script and different settings.

One change I made to MapUnitPolys - I added an integer field MapUnit2 so I could make subtypes to create a drop-down list of mapping units for ease when building polygons.
You'll see the polygons are totally arbitrary, just to have something to test the cross section tool with.

Kelin

https://www.dropbox.com/scl/fi/qtbwuf3172lyqprn7p6yp/Alluvial_Fan_Mapping.gdb.zip?rlkey=0zr3ypro0a2sku210p0vveqig&dl=0


________________________________

**Reply by ethoms-usgs** [reply]
So, it ran for me fine. Here's how I set up the map and the parameter form:

![Screenshot 2024-08-28 135315](https://github.com/user-attachments/assets/13d88eaf-0631-4c05-8d4a-291bab705e6c)

Is that essentially the same as what you have? I was mistaken before when I said that the change I made to the script I posted ensured the tool knew the full path to the feature classes and the fact that the tool runs (partially?) when you check 'Project all features' suggests to me that is still the problem, but I don't know why.

**Comment by kxwhipple** [comment]
Ha!  Of course it runs for you.  Isnt that how it always is.
Those are the same parameters I was trying.  If I select just those two feature classes to project, I get this error almost immediately:

  File "C:\GEMS_USGS_Mapping\gems-tools-pro-xs-update\Scripts\GeMS_ProjectCrossSectionData_AGP2.py", line 489, in <module>
﻿    desc = arcpy.da.Describe(fc)
﻿           ^^^^^^^^^^^^^^^^^^^^^
﻿ValueError: Object: Error in accessing describe

But it runs (completely) for me if and only if I select Project all features in the GeologicMap ... So its fine, that's the natural default anyway.

Still the error suggests some sensitivity to exactly the file structure on my PC, or what geodatabase I use for Scratch workspace?

Thanks again for all your help.
________________________________

**Reply by ethoms-usgs** [reply]
Ugh, I know. Frustrating.

The only thing I can think of to try at this point is to simplify everything. Make a smaller gdb, save it on a short local path with no spaces (same for your default.gdb), and make a new map with only the layers you need.

**Comment by kxwhipple** [comment]
﻿Indeed I was thinking the same thing.  My organization could be cleaner and that may make the difference. I will test some variations.

But bottom line it is indeed working, at least if I use those two options.

Kelin

Sent from my iPhone

On Aug 29, 2024, at 9:08 AM, Evan Thoms ***@***.***> wrote:

﻿

Ugh, I know. Frustrating.

The only thing I can think of to try at this point is to simplify everything. Make a smaller gdb, save it on a short local path with no spaces (same for your default.gdb), and make a new map with only the layers you need.
